cnd/src/swap_protocols/halight/connector.rs

```diff
@@ -432,6 +513,56 @@ fn client(certificate: &Certificate, macaroon: &Macaroon) -> Result<reqwest::Cli
     Ok(client)
 }
 
+pub fn deserialize_amount<'de, D>(deserializer: D) -> Result<asset::Bitcoin, D::Error>
```

Comment:
We already implemented this deserialisations somewhere else, can we try to re-use the code instead?

Reply:
We haven't interacted with LN before so I wouldn't want to re-use API-specific code that just happens to be the same.
There is no super harmful duplication here as we just delegate to `from_sat` for the crucial part.
It is annoying that we have to do this, but if we want to avoid it then the solution shouldn't be: "just use the HTTP API serialization code".